code/modules/client/client_procs.dm

```diff
@@ -158,6 +158,9 @@ GLOBAL_LIST_EMPTY(external_rsc_urls)
 	return var_name != NAMEOF(src, holder) && ..()
 
 /client/New(TopicData)
+	//hold the connection until a certain point in world/Startup()
+	UNTIL(GLOB.log_directory)
```

Comment:
As well as this

Reply:
are you even allowed to sleep in client login

Reply:
yes, ipintel does it, it has predictable results, the opening of the interface is delayed while the server sleeps.

Also client/New doesn't happen on reboots until world/New() exits or calls ..() iirc. somebody test this

Reply:
Incorrect it seems

```dm
#define UNTIL(X) while(!(X)) sleep(1)

var/world_ready = FALSE

/proc/Log(msg)
	world.log << "[world.time]: [msg]"

/client/New()

	Log("client/New() start: [ckey]")
	
	UNTIL(world_ready)
	
	Log("client/New() done sleeping: [ckey]")
	
	. = ..()
	
	Log("client/New() complete: [ckey]")
	
/world/New()

	Log("world/New() start")
	sleep(30)
	Log("world/New() done sleeping")
	Startup()

/world/proc/Startup()
	set waitfor = FALSE
	
	Log("world/Startup() start")
	sleep(30)
	Log("world/Startup() done sleeping")
	
	world_ready = TRUE
	
	sleep(30)
	Log("world/Startup() complete")
	



/client/verb/do_reboot()
	set name = "Reboot"
	
	Log("client [ckey] used reboot")
	world.Reboot()
```

```
Sat Jun 09 14:13:17 2018
World opened on network port 12345.
Welcome BYOND! (5.0 Beta Version 512.1423)
0: world/New() start
30: world/New() done sleeping
30: world/Startup() start
60: world/Startup() done sleeping
90: world/Startup() complete
92: client/New() start: guest3611034376
92: client/New() done sleeping: guest3611034376
92: client/New() complete: guest3611034376
156: client guest3611034376 used reboot
Rebooted server.
0: world/New() start
3: client/New() start: guest3611034376
30: world/New() done sleeping
30: world/Startup() start
60: world/Startup() done sleeping
60: client/New() done sleeping: guest3611034376
60: client/New() complete: guest3611034376
90: world/Startup() complete
```